Dockerfile.release

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright 2025 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Debian 12 (Bookworm) image to build binary for distroless/static-debian12
+FROM golang:1.24.2@sha256:fb224f950b0dfb889f8f1122bf3dfc21976735ccf983b73a1e6e014215a272f5 AS builder
+ENV APP_ROOT=/opt/app-root
+ENV GOPATH=$APP_ROOT
+
+WORKDIR $APP_ROOT/src/
+ADD go.mod go.sum $APP_ROOT/src/
+RUN go mod download
+
+# Add source code
+ADD ./cmd/ $APP_ROOT/src/cmd/
+ADD ./pkg/ $APP_ROOT/src/pkg/
+
+ARG SERVER_LDFLAGS
+RUN go build -ldflags "${SERVER_LDFLAGS}" ./cmd/rekor-server
+RUN CGO_ENABLED=0 go build -gcflags "all=-N -l" -ldflags "${SERVER_LDFLAGS}" -o rekor-server ./cmd/rekor-server
+
+# Multi-Stage production build
+FROM gcr.io/distroless/static-debian12:nonroot@sha256:c0f429e16b13e583da7e5a6ec20dd656d325d88e6819cafe0adb0828976529dc AS deploy
```

Comment:
Could we just update the original dockerfile to have this minimal base? These dockerfiles are almost the same, it seems like we could just use the same one. Otherwise we'll have to be conscious of keeping them in sync with each other. It seems like we don't bother with separate dockerfiles for the other services?

Reply:
For other services, we've had one ko file which is for production containers and one Dockerfile only for development. I'm fine with ko but have a preference towards being explicit with the steps for our container builds, which is why there are now two dockerfiles.

There is a difference between these, as the other dockerfile has a third stage of installing dlv. We could add an argument to skip this stage when set. WDYT?

Reply:
Having an argument to skip the debug stage sgtm

Reply:
Updated, lemme know what you think! We do need different base images for the prod vs debug containers since the debug container needs a shell, so I added multiple stages.